rust/datafusion/src/physical_plan/projection.rs

```diff
@@ -114,14 +117,33 @@ impl ExecutionPlan for ProjectionExec {
     }
 
     async fn execute(&self, partition: usize) -> Result<SendableRecordBatchStream> {
+        let input = if self.input.as_any().downcast_ref::<EmptyExec>().is_some() {
```

Comment:
I like the simplicity of this special case for execution. 

However, I think it might be slightly confusing that `EmptyExec` will produce different output depending on where in the plan it is. Thus you can't reason about what EmptyExec will produce by itself (you need to know how it is used).

I wonder if we could add a field such as  `produce_one_row: bool`  to `EmptyExec`  and call `placeholder_stream` in `EmptyExec::execute`[here](https://github.com/apache/arrow/blob/master/rust/datafusion/src/physical_plan/empty.rs#L81)

However, this is a stylistic comment and we can always make that improvement later on as another PR

Reply:
I agree with @alamb : this design does not scale as we end up with `if` statements scattered around the code.

Reply:
I add this field in commit ef09e12.

This flag is set in logical plan phase, `from_join_to_plan()`. If ast has no `FROM` clause then an `EmptyRelation` node will be added with `produce_one_row` set to true. And then comes to exec phase, `EmptyExec` will produce a placeholder row.

Maybe this is better? :smiley:

Reply:
Maybe a separate `OneRowExec` and corresponding logical `OneRowRelation` would be even cleaner, but that is probably nitpicking.

Reply:
@jhorstmann thanks for your suggestion!

Hi @alamb @jorgecarleitao, what do you think of this? I'm OK with both. But in my opinion, I prefer the first one because I think it's enough for this situation.